classes/ActionScheduler_Versions.php

```diff
@@ -122,65 +128,24 @@ public static function initialize_latest_version() {
 	 *         'name' => 'Name',
 	 *     ]
 	 *
+	 * @deprecated 3.9.1 Use ActionScheduler_SystemInformation::active_source().
+	 *
 	 * @return array
 	 */
 	public function active_source(): array {
-		$file         = __FILE__;
-		$dir          = __DIR__;
-		$plugins      = get_plugins();
-		$plugin_files = array_keys( $plugins );
-
-		foreach ( $plugin_files as $plugin_file ) {
-			$plugin_path = trailingslashit( WP_PLUGIN_DIR ) . dirname( $plugin_file );
-			$plugin_file = trailingslashit( WP_PLUGIN_DIR ) . $plugin_file;
-
-			if ( 0 !== strpos( dirname( $dir ), $plugin_path ) ) {
-				continue;
-			}
-
-			$plugin_data = get_plugin_data( $plugin_file );
-
-			if ( ! is_array( $plugin_data ) || empty( $plugin_data['Name'] ) ) {
-				continue;
-			}
-
-			return array(
-				'type' => 'plugin',
-				'name' => $plugin_data['Name'],
-			);
-		}
-
-		$themes = (array) search_theme_directories();
-
-		foreach ( $themes as $slug => $data ) {
-			$needle = trailingslashit( $data['theme_root'] ) . $slug . '/';
-
-			if ( 0 !== strpos( $file, $needle ) ) {
-				continue;
-			}
-
-			$theme = wp_get_theme( $slug );
-
-			if ( ! is_object( $theme ) || ! is_a( $theme, \WP_Theme::class ) ) {
-				continue;
-			}
-
-			return array(
-				'type' => 'theme',
-				// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-				'name' => $theme->Name,
-			);
-		}
-
-		return array();
+		_deprecated_function( __METHOD__, '3.9.1', 'ActionScheduler_SystemInformation::active_source()' );
+		return ActionScheduler_SystemInformation::active_source();
 	}
 
 	/**
 	 * Returns the directory path for the currently active installation of Action Scheduler.
 	 *
+	 * @deprecated 3.9.2 Use ActionScheduler_SystemInformation::active_source_path().
```

Comment:
Is this supposed to be 3.9.**2**?

Reply:
Yes: but thank you for flagging as, unless I'm having a moment's confusion, some of the other references to 3.9.1 should also be updated.

Reply:
@dwainm. Made some updates. Summarizing:

- The problem was introduced in 3.9.1 (current stable release).
- Assuming for the moment this set of changes ship in 3.9.2, then that is the version number to use for the deprecations.
- And in case of any doubt, [this specific version number](https://github.com/woocommerce/action-scheduler/blob/412f067449f98e15048af21d3052205c6fb4a0b4/classes/ActionScheduler_Versions.php#L62) (which is not a deprecation) remains at 3.9.1.

I hope that makes sense, but let me know if this feels incorrect to you.

Reply:
Yes, it makes sense for this to be targeting 3.9.2 as 3.9.1 is out already. We just want to catch any updates early so folks don't even think about 1 and jump straight to two.